src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -155,6 +155,11 @@ class TaskReplaceActor(
           logger.info(s"Killing old ${nextOldInstance.instanceId}")
       }
 
+      if (runSpec.isResident) {
+        instanceTracker.goalStopped(nextOldInstance.instanceId)
```

Comment:
Why not:

```
instanceTracker.updateGoal(nextOldInstance.instanceId, Goal.Stopped)
```

?

Do we need a separate method for each goal state or sufficient to pass the appropriate enum value?

Reply:
+1. The scheduler could have this
```scala
trait Scheduler {
  def run(instance: Instance*) : Future[Done]
  def stop(instance: Instance*) : Future[Done]
  def decommission(instance: Instance*) : Future[Done]
}
```

its implementation would then call `tracker.update` or `tracker.updateGoal`.